.github/workflows/main.yml

```diff
@@ -195,6 +195,12 @@ jobs:
     steps:
       - name: git checkout
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
Interesting, so without this it won't fetch all commits?

Reply:
yeah - if you dont have this it cant properly apply the rebase and so will produce a bunch of merge conflicts.

Unfortunate cause this is obvs less efficient but at least it seems to be very fast still :)

Reply:
Thanks for the explanation, TIL 💡 :)

Reply:
IIRC the CI won't run if there's a merge conflict

Reply:
indeed - but what I mean here is that if you dont do `fetch-depth: 0`, then it will always _think_ that there is a merge conflict even if there is none